### DISCUSSION TITLE
Is anyone interested in developing together STM32 SAI Driver

### DISCUSSION THREAD

**Discussion by mariopaja** [zephyrproject-rtos/zephyr#82469] [Q&A]
At the moment in Zephyr is only I2S driver implemented for MCUs that have a muxed SPI/I2S, but this approach leaves outside a lot of MCUs, in our case U5xx Series 

Therefore, we are looking at the possibility of implementing a STM32 SAI driver. 

In this case I want to find the opportunity to ask if there other developers that have the same interest or need so we can work together in this development

**Comment by mariopaja** [comment]
At the moment I have a first working draft ready which I will push soon.

The SAI at the moment is interrupt driven. Until now I was not able to implement DMA transfers. I will appreciate any help in this regard 🙏🏻

**Comment by chrisschir** [comment]
Hey @mariopaja ,
I have developed a driver for the U5 that works but only for my specific problem at the moment. E.g. the SAI blocks a configured fixed for input or output and I have only used it for I2S at the moment. The DMA works.
Do you have a branch for this project where I can have a look at your current status?

**Reply by mariopaja** [reply]
Hi @chrisschir,

Thank you for your reply.
It has been been a bit that I haven't worked on this topic due to other projects. I was planning to revisit it soon.

Here is a draft that I have published: https://github.com/zephyrproject-rtos/zephyr/pull/82945

**Reply by chrisschir** [reply]
Hi @mariopaja,
I think this topic can be closed?

**Reply by mariopaja** [reply]
> Hi @mariopaja,
> 
> I think this topic can be closed?

Yes it is :)
